spinnaker_graph_front_end/__init__.py

```diff
@@ -55,19 +83,23 @@
 
 
 def setup(hostname=None, graph_label=None, model_binary_module=None,
-          model_binary_folder=None, database_socket_addresses=None,
+          model_binary_folder=None, database_socket_addresses=(),
           user_dsg_algorithm=None, n_chips_required=None,
-          n_boards_required=None, extra_pre_run_algorithms=None,
-          extra_post_run_algorithms=None,
+          n_boards_required=None, extra_pre_run_algorithms=(),
+          extra_post_run_algorithms=(),
```

Comment:
Do we know for sure that changing these to not None works?

Reply:
The underlying actual requirement is for an iterable. I chased down what we were doing with it (inserting into various internally-defined lists) and checked that the types of insertion used actually worked with a general iterator (e.g., a one-shot iterator such as a generator).

Reply:
OK, fair enough